### PR TITLE
[dev] [No Bug/PBI] Fix `data-testid` issue for `<DataGroups>`

### PR DESCRIPTION
**Pull request by groberts314**
This fixes the following warning (and makes the `data-testid` actually useful!)
```
react_devtools_backend.js:4061 Warning: React does not recognize the `dataTestId` prop on a DOM element.
If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `datatestid` instead.
If you accidentally passed it from a parent component, remove it from the DOM element.
    in div (created by DataGroups)
    in DataGroups (created by PageDataGroups)
    in PageDataGroups (created by [Some Component Using Page Data Groups])
    ...
```
Also adds a unit test fixture for `<DataGroups>` and fixes a typo I noticed in `<DemoPage>` in the docs.

### CODE REVIEW

**Review by fantaJinMode — COMMENTED** [review]
LGTM 👍 Apart from one repetitive test.

File: src/dataDisplay/dataGroups/__test__/dataGroups.test.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        render(
+            <DataGroups {...props} />,
+        );
+
+        const rootDataGroupsContainer = screen.queryByTestId(props.dataTestId);
+        expect(rootDataGroupsContainer).toBeInTheDocument();
+        expect(rootDataGroupsContainer).toHaveClass(className);
+        expect(screen.queryByText('Column 1')).toBeInTheDocument();
+        expect(screen.queryByText('Column 2')).toBeInTheDocument();
+        expect(screen.queryByText('Column 1')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (groberts314) [reply src/dataDisplay/dataGroups/__test__/dataGroups.test.js]: Fixed: 68c719f
Thanks for catching that!

File: docs/src/layout/page/demo/demo.jsx [hunk 2]
```diff
@@ -1137,7 +1137,7 @@ class PageDemo extends React.PureComponent {
                                 className={classes.dataGroupsContainer}
                             >
                                 <Page.DataGroups
-                                    columns={mainCoulumn}
+                                    columns={mainColumns}
```
Comment (cbrewer1689) [inline docs/src/layout/page/demo/demo.jsx:1140]: I personally like the misspelling, it shows how human I am. Kidding.

File: src/surfaces/actionBar/actionBar.jsx [hunk 3]
```diff
@@ -308,7 +308,7 @@ class ActionBar extends React.PureComponent {
                                                 <Button
                                                     color={button.color}
                                                     designVersion={1}
-                                                    disabled={!!button.disabled}
+                                                    disable={!!button.disable || !!button.disabled}
```
Comment (cbrewer1689) [inline src/surfaces/actionBar/actionBar.jsx:311]: I wonder if we should just keep this `disabled`. One reason, designVersion `1` supports it and two, designVersion `2` requires it. Up to you.

**Review by cbrewer1689 — APPROVED** [review]

File: src/surfaces/actionBar/actionBar.jsx (hunk 3 above)
Comment (groberts314) [reply src/surfaces/actionBar/actionBar.jsx:311]: I'm good either way.  I've always favored `disabled` for consistency with native DOM and most standards I am aware of.
I just want whatever:
a) won't break stuff, and
b) won't log too many warnings! =)

Comment (cbrewer1689) [reply src/surfaces/actionBar/actionBar.jsx:311]: B convinces me. I hate logs/warnings.

**Pull request merged**